### PR TITLE
clean up the code to change inventory_index_id arg to type int64 everywhere except at the parser level

### LINKED ISSUE #1559: Purge Inventory: TypeError: sequence item 0: expected string, long found
```
forseti-client-vm-4330:~$ forseti inventory purge 0
```

```
May 18 06:23:05 forseti-server-vm-4330 forseti_server[15232]: INFO:google.cloud.forseti.services.inventory.inventory:Cut-off datetime to start purging is: 2018-05-18 06:23:05.510012
May 18 06:23:05 forseti-server-vm-4330 forseti_server[15232]: ERROR:root:Exception calling application: sequence item 0: expected string, long found
May 18 06:23:05 forseti-server-vm-4330 forseti_server[15232]: Traceback (most recent call last):
May 18 06:23:05 forseti-server-vm-4330 forseti_server[15232]:   File "/usr/local/lib/python2.7/dist-packages/grpc/_server.py", line 385, in _call_behavior
May 18 06:23:05 forseti-server-vm-4330 forseti_server[15232]:     return behavior(argument, context), True
May 18 06:23:05 forseti-server-vm-4330 forseti_server[15232]:   File "/usr/local/lib/python2.7/dist-packages/forseti_security-2.0.0-py2.7.egg/google/cloud/forseti/services/inventory/service.py", line 168, in Purge
May 18 06:23:05 forseti-server-vm-4330 forseti_server[15232]:     result = self.inventory.purge(request.retention_days)
May 18 06:23:05 forseti-server-vm-4330 forseti_server[15232]:   File "/usr/local/lib/python2.7/dist-packages/forseti_security-2.0.0-py2.7.egg/google/cloud/forseti/services/inventory/inventory.py", line 358, in purge
May 18 06:23:05 forseti-server-vm-4330 forseti_server[15232]:     purged_inventory_indexes_as_str = ', '.join(purged_inventory_indexes)
May 18 06:23:05 forseti-server-vm-4330 forseti_server[15232]: TypeError: sequence item 0: expected string, long found
```

### PR DESCRIPTION
**Pull request by joecheuk**
Resolves #1559, #1560

### CODE REVIEW

File: google/cloud/forseti/services/client.py [hunk 1]
```diff
@@ -394,7 +394,7 @@ def delete(self, inventory_index_id):
         """Delete an inventory.
 
         Args:
-            inventory_index_id (str): the index id of the inventory to delete.
+            inventory_index_id (int): the index id of the inventory to delete.
```
Comment (ahoying) [inline google/cloud/forseti/services/client.py]: Nit: int64 (to be consistent)

File: google/cloud/forseti/services/inventory/storage.py [hunk 2]
```diff
@@ -521,12 +522,12 @@ def initialize(engine):
...
     """Inventory storage used during creation."""
 
-    def __init__(self, session, existing_id=None, readonly=False):
+    def __init__(self, session, existing_id=0, readonly=False):
         """Initialize
 
         Args:
             session (object): db session
-            existing_id (str): The inventory id if wants to open an existing one
+            existing_id (int64): The inventory id if wants to open an existing one
```
Comment (ahoying) [inline google/cloud/forseti/services/inventory/storage.py]: Putting comment here for lack of a better place:

This should probably not return a string any longer either?

https://github.com/GoogleCloudPlatform/forseti-security/blob/1a8a62cd7f7fbaf4ac89d4a7c1349cf2c722d1d0/google/cloud/forseti/services/inventory/storage.py#L490

File: google/cloud/forseti/services/cli.py [hunk 3]
```diff
@@ -788,7 +788,7 @@ def do_create_model():
         """Create a model."""
         result = client.new_model('inventory',
                                   config.name,
-                                  config.inventory_index_id,
+                                  long(config.inventory_index_id),
```
Comment (ahoying) [inline google/cloud/forseti/services/cli.py]: Python will auto-convert an int to a long if it's greater than maxint, yeah? So I don't think we need to specify long here, int is fine.

```
>>> sys.maxint
9223372036854775807
>>> x = sys.maxint
>>> x
9223372036854775807
>>> x = sys.maxint + 1
>>> x
9223372036854775808L
```

Comment (joecheuk) [reply google/cloud/forseti/services/cli.py]: Updated.

File: google/cloud/forseti/services/client.py (hunk 1 above)
Comment (joecheuk) [reply google/cloud/forseti/services/client.py]: Updated.

File: google/cloud/forseti/services/inventory/storage.py (hunk 2 above)
Comment (joecheuk) [reply google/cloud/forseti/services/inventory/storage.py]: This method is used in notifier, I have updated notifier to use inventory_index_id as int, too. PTAL again!

**Pull request merged**